patches/minecraft/net/minecraft/world/item/ItemStack.java.patch

```diff
@@ -205,6 +205,16 @@
        return list;
     }
  
+@@ -736,7 +_,8 @@
+    }
+ 
+    private int m_41618_() {
+-      return this.m_41782_() && this.f_41590_.m_128425_("HideFlags", 99) ? this.f_41590_.m_128451_("HideFlags") : 0;
++      int stackFlags = this.m_41782_() && this.f_41590_.m_128425_("HideFlags", 99) ? this.f_41590_.m_128451_("HideFlags") : 0;
++      return stackFlags | net.minecraftforge.common.ForgeHooks.getDefaultHideFlags(this);
+    }
```

Comment:
Personally, I think this should be JUST replacing the 0 value. Because if you set the HideFlags in the NBT then it should be the end-all. They should not be 'in addition to the default'.

Also, I get what you're going for with the thing being a set, however, performance wise I think it'd be better to have the value return a int. So it doesn't need to recalculate the values all the time. And it's more likely that a modder will screw up creating a new set every call then it is they will screw up `return CAN_PLACE.getMask() | DYE.getMask()`